lib/mongodb/native_mongodb.nit

```diff
@@ -506,7 +506,14 @@ extern class NativeMongoCursor `{ mongoc_cursor_t* `}
 	# Wrapper for `mongoc_cursor_current()`.
 	#
 	# Fetches the cursors current document or NULL if there has been an error.
-	fun current: NativeBSON `{ return (bson_t*) mongoc_cursor_current(self); `}
+	fun current: NativeBSON `{
+		// As said in documentation, BSON objects should not be freed manually.
+		bson_t* bson = (bson_t*) mongoc_cursor_current(self);
+		// Copy BSON so we can let the GC free it automatically.
+		bson_t* ret = malloc(sizeof(bson_t));
+		memcpy(ret, bson, sizeof(bson_t));
```

Comment:
this might be fragile if:
* there is sub-data in the struct (that will fail if the memcpy is shallow); it might be the case since json is by definition a recursive data-structure.
* the `bson_t` is used as a C abstract data type where the runtime data type and allocated size depends on the specific real piece of data (that will fail since you use the sizeof(bson_t) and not the real and unknown size); it might be the case since json can store heterogeneous things.